packages/payments-plugin/package.json

```diff
@@ -29,7 +30,8 @@
         "stripe": "13.x"
     },
     "dependencies": {
-        "currency.js": "2.0.4"
+        "currency.js": "2.0.4",
+        "msw": "^2.4.4"
```

Comment:
I can not use nock with the native fetch api that is used within the paypal integration.
See this issue for more information: https://github.com/nock/nock/issues/2397

The fetch Api was introduced in NodeJS v18, so there shouldn't be any compatibility problems I think.

Reply:
See https://www.npmjs.com/package/msw for more information

Reply:
`nock` does support the native `fetch` in the beta version `npm I nock@beta.` It's stable and has been in beta for a few months now.

Reply:
Thank you, I will try it again with the beta release and change it for our tests if necessary.

Reply:
Switched to nock and updated to beta version.

Reply:
Updating the nock version broke other test cases. @mikicho maybe you can take a brief look at the test case in hopes that you see the issue directly: https://github.com/vendure-ecommerce/vendure/blob/dcf7b0a023532e815b7aff0995389db4f9ef6bb7/packages/payments-plugin/e2e/stripe-payment.e2e-spec.ts#L135

I will try to locate the issue myself, just thought you may see the issue directly :)

Reply:
Seems ok to me.. what error do you get?

Reply:
I could not get an error for now, it just hits the timeout of the test case. But I found the exact location where the timeout happens, which is here: 
https://github.com/vendure-ecommerce/vendure/blob/dcf7b0a023532e815b7aff0995389db4f9ef6bb7/packages/payments-plugin/src/stripe/stripe.service.ts#L173

Reply:
To be more precise here: https://github.com/stripe/stripe-node/blob/768206df1b9d025853d5e34e6046beb502effdd9/src/net/NodeHttpClient.ts#L83

The callback of `req.on('response')` won't be called after upgrading to `nock@14.0.0-beta.15` and with `stripe@17.2.1`. I think I will create an issue at nocks repository.